src/system.ts

```diff
@@ -81,39 +81,32 @@ const chainSource = new ChainSource({
   arweaveBaseUrl: config.ARWEAVE_URL,
 });
 
-// Attempt to read the start height and epoch block length from the contract
-let epochZeroStartHeigth = START_HEIGHT;
-let epochBlockLength = EPOCH_BLOCK_LENGTH;
-try {
-  const resp = await got
-    .get(
-      `${config.CONTRACT_CACHE_URL}/v1/contract/${config.CONTRACT_ID}/read/epoch`,
-    )
-    .json<any>();
-  epochZeroStartHeigth = +(resp?.result?.epochZeroStartHeight ?? START_HEIGHT);
-  log.info(
-    `Got epoch zero start height from contract: ${epochZeroStartHeigth}`,
-    {
-      epochZeroStartHeigth,
-    },
-  );
-  epochBlockLength = +(resp?.result?.epochBlockLength ?? EPOCH_BLOCK_LENGTH);
-  log.info(`Got epoch block length from contract: ${epochBlockLength}`, {
-    epochBlockLength,
+const networkContract = new ArIO({
+  contractTxId: config.CONTRACT_ID,
+});
+
+// Attempt to read the start height and epoch block length from the contract - default to constants if it fails
+const { epochZeroStartHeight, epochBlockLength } = await networkContract
+  .getCurrentEpoch()
+  .catch((error: unknown) => {
+    log.error(
+      `Unable to get start height from contract cache - using default values`,
+      {
+        error: error instanceof Error ? error.message : error,
```

Comment:
It's less typesafe, but I still prefer using `any` for errors and then explicitly pulling out `message` and `stack` if they're set since it handles a wider range of possible values:
```
{
  message: error?.message,
  stack: stack?.stack,
}
```
I'm open to counter arguments why checking the type is a better pattern though.

Reply:
no objections - the stack will also help highlight any issues in the SDK

Reply:
> in your example did you intend error?.stack?

Whoops, yes. :sweat_smile: